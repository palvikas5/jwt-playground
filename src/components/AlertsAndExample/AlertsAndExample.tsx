import React from 'react';
import Image from 'next/image';
import { useRouter } from 'next/router';
import Button from '../_core/Button/Button';
import constants from '../../utils/constants';
import {
  StyledAlertsAndExample,
  StyledTokenInvalidWrapper,
  StyledInvalidInput,
} from './alertsAndExample.style';

export interface AlertsAndExampleProps {
  isTokenInvalid: boolean;
}

const AlertsAndExample = ({ isTokenInvalid }: AlertsAndExampleProps) => {
  const router = useRouter();

  const navigateToExample = () => {
    router.push(`/?value=${constants.defaultRawToken}`);
  };

  return (
    <StyledAlertsAndExample>
      {isTokenInvalid && (
        <StyledTokenInvalidWrapper>
          <Image src='/warning.svg' alt='warning' width='20' height='20' />
          <StyledInvalidInput>Invalid input!</StyledInvalidInput>
        </StyledTokenInvalidWrapper>
      )}
      <Button onClick={navigateToExample}>Try out an Example</Button>
    </StyledAlertsAndExample>
  );
};

export { AlertsAndExample };
