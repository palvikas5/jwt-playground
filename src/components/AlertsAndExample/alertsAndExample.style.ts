import styled from 'styled-components';
import { device } from '../../utils/device';

const StyledAlertsAndExample = styled.div`
  display: flex;
  flex-direction: column;

  @media ${device.mediumAndAbove} {
    flex-direction: row;
    align-items: center;
    justify-content: flex-end;
  }
`;

const StyledTokenInvalidWrapper = styled.div`
  display: flex;
  flex: 1;
  align-items: center;
  font-weight: bold;
  color: #d84c1a;
  background: #fce0cf;
  padding: 10px;
  border-radius: 3px;
  margin-bottom: 10px;

  @media ${device.mediumAndAbove} {
    margin-bottom: 0;
    margin-right: 10px;
  }
`;

const StyledInvalidInput = styled.div`
  margin-left: 10px;
`;

export {
  StyledAlertsAndExample,
  StyledInvalidInput,
  StyledTokenInvalidWrapper,
};
