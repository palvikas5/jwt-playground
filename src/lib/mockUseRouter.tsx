const useRouter = jest.spyOn(require('next/router'), 'useRouter');

function mockUseRouter(props: {
  route?: string;
  pathname?: string;
  query?: string;
  asPath?: string;
  push?: (path: string) => void;
}) {
  useRouter.mockImplementationOnce(() => ({
    route: props.route,
    pathname: props.pathname,
    query: props.query,
    asPath: props.asPath,
    push: props.push,
  }));
}

export { mockUseRouter };
