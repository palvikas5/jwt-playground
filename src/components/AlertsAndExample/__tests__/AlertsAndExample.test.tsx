import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { AlertsAndExample, AlertsAndExampleProps } from '../AlertsAndExample';
import { mockUseRouter } from '../../../lib/mockUseRouter';

const push = jest.fn();

const setup = (props: AlertsAndExampleProps) => {
  mockUseRouter({ push });

  return render(<AlertsAndExample {...props} />);
};

describe('alertsAndExample', () => {
  it('should render successfully without invalid alert', () => {
    expect.assertions(1);

    const { container } = setup({ isTokenInvalid: false });
    expect(container).toMatchSnapshot();
  });

  it('should render successfully with invalid token alert', () => {
    expect.assertions(1);

    const { container } = setup({ isTokenInvalid: true });
    expect(container).toMatchSnapshot();
  });

  it('should trigger router change on click of example button', () => {
    expect.assertions(1);

    const { getByText } = setup({ isTokenInvalid: true });
    const button = getByText('Try out an Example');
    fireEvent.click(button);
    expect(push).toHaveBeenCalledWith(
      '/?value=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c',
    );
  });
});
