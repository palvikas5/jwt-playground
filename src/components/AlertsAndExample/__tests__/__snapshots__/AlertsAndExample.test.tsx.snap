// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alertsAndExample should render successfully with invalid token alert 1`] = `
.c3 {
  display: block;
  color: white;
  background-color: #3e8efa;
  text-transform: uppercase;
  padding: 12px 15px;
  border: none;
  border-radius: 3px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: bold;
  color: #d84c1a;
  background: #fce0cf;
  padding: 10px;
  border-radius: 3px;
  margin-bottom: 10px;
}

.c2 {
  margin-left: 10px;
}

@media (min-width:768px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
  }
}

@media (min-width:768px) {
  .c1 {
    margin-bottom: 0;
    margin-right: 10px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="box-sizing: border-box; display: block; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjAiIGhlaWdodD0iMjAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
            style="max-width: 100%; display: block;"
          />
        </div>
        <img
          alt="warning"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </div>
      <div
        class="c2"
      >
        Invalid input!
      </div>
    </div>
    <button
      class="c3"
      type="button"
    >
      Try out an Example
    </button>
  </div>
</div>
`;

exports[`alertsAndExample should render successfully without invalid alert 1`] = `
.c1 {
  display: block;
  color: white;
  background-color: #3e8efa;
  text-transform: uppercase;
  padding: 12px 15px;
  border: none;
  border-radius: 3px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media (min-width:768px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
  }
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1"
      type="button"
    >
      Try out an Example
    </button>
  </div>
</div>
`;
